util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/dialect/ElasticsearchTestDialect.java

```diff
@@ -29,8 +33,26 @@ public static ElasticsearchVersion getActualVersion() {
 			ElasticsearchDistributionName distribution =
 					ElasticsearchDistributionName.of( System.getProperty(
 							"org.hibernate.search.integrationtest.backend.elasticsearch.distribution" ) );
-			String versionString = System.getProperty(
-					"org.hibernate.search.integrationtest.backend.elasticsearch.version" );
+			String versionString = "";
+
+			if ( distribution != ElasticsearchDistributionName.AMAZON_OPENSEARCH_SERVERLESS ) {
+				try ( TestElasticsearchClient client = new TestElasticsearchClient() ) {
+					client.open( new TestConfigurationProvider(), Optional.empty() );
+					// This will probably use org.hibernate.search.backend.elasticsearch.client.impl.ElasticsearchClientUtils#tryGetElasticsearchVersion
+					versionString = client.getActualVersion();
+					if ( versionString != null ) {
+						// If we got a snapshot version back from the actual cluster we want to drop that qualifier part of version.
+						// Qualifiers are not allowed when comparing versions with a comparator from this class
+						// (see this#compare(ElasticsearchVersion a, ElasticsearchVersion b, int defaultInt)).
+						versionString = versionString.substring( 0, versionString.indexOf( '-' ) );
+					}
+				}
+				catch (IOException e) {
+					versionString =
+							System.getProperty( "org.hibernate.search.integrationtest.backend.elasticsearch.version", "" );
```

Comment:
So if this property is set, it's ignored by default in favor of automatic detection? Not a great UX xD

Either:

* you're sure this automatic detecting will always work and feel free to just remove `org.hibernate.search.integrationtest.backend.elasticsearch.version` everywhere
* or you're not and we'd better take `org.hibernate.search.integrationtest.backend.elasticsearch.version` into account when it's set.

In any case, you should probably update the documentation (CONTRIBUTING.md or MAINTAINERS.md) accordingly :)

Reply:
Thanks.

> In any case, you should probably update the documentation (CONTRIBUTING.md or MAINTAINERS.md) accordingly :)

Ok so I see `test.elasticsearch.version` is still needed regardless, so let's ignore that suggestion.

Reply:
Yeah... unless we want to pass the version of es container to start in some other way....